automation/check-patch.e2e-multus-functests.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash -xe
+
+# This script should be able to execute multus
+# functional tests against Kubernetes cluster with
+# CNAO built with latest changes, on any
+# environment with basic dependencies listed in
+# check-patch.packages installed and docker running.
+#
+# yum -y install automation/check-patch.packages
+# automation/check-patch.e2e-multus-functests.sh
+
+teardown() {
+    make cluster-down
+    rm -rf "${TMP_COMPONENT_PATH}"
+}
+
+function __get-tools() {
+    if [ ! -d bin ]; then
+        mkdir bin
+    fi
+
+    echo "install jq"
+    curl -Lo ./bin/jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64
+    chmod +x ./bin/jq
+}
+
+function __prepare-test-environment() {
+    __get-tools
+
+    echo "Deploy cni-plugins pod"
+    CNI_PLUGIN_YAML="cni-install.yml"
+    kubectl create -f ${CNI_PLUGIN_YAML}
+    kubectl -n kube-system wait --for=condition=ready -l name=cni-plugins pod --timeout=300s
+
+    TEST_YAML="*macvlan1.yml"
+    # matching the test nodes to the CNAO cluster nodes
+    sed -i 's/kind-worker$/node01/g' ${TEST_YAML}
+    sed -i 's/kind-worker2$/node02/g' ${TEST_YAML}
+
+    echo "Set eth1 NIC to up using NodeNetworkConfigurationPolicy"
+    cat <<EOF | kubectl apply -f -
+apiVersion: nmstate.io/v1beta1
+kind: NodeNetworkConfigurationPolicy
+metadata:
+  name: eth1-policy
+spec:
+  desiredState:
+    interfaces:
+    - name: eth1
+      description: Configuring eth1 on all nodes
+      type: ethernet
+      state: up
+      ipv4:
+        dhcp: true
+        enabled: true
+EOF
+}
+
+function __remove-test-manifests() {
+    echo "Remove test manifests"
+    kubectl delete -f ${TEST_YAML}
+    kubectl delete -f ${CNI_PLUGIN_YAML}
+}
+
+main() {
+    # Setup CNAO and artifacts temp directory
+    source automation/check-patch.setup.sh
+    cd ${TMP_PROJECT_PATH}
+
+    # Spin-up ephemeral cluster with latest CNAO, 2 nodes and secondary NIC needed for test
+    export KUBEVIRT_NUM_NODES=2
+    export KUBEVIRT_NUM_SECONDARY_NICS=1
+    # this script also exports KUBECONFIG, and fetch $COMPONENT repository
+    COMPONENT="multus" source automation/components-functests.setup.sh
+
+    trap teardown EXIT
```

Comment:
aint the trap should be before line 74 ?
you might want to add SIGINT

Reply:
no, only `automation/components-functests.setup.sh` creates the cluster.
IIUC only EXIT is required.

Reply:
ctrl c, will work just if SIGINT is used, from what i remember
up to you

Reply:
but if 
no, only automation/components-functests.setup.sh creates the cluster.
failed, u do want to cluster-down as some containers might already exists

Reply:
in bash EXIT works also for `ctrl c`,(implied [here](https://unix.stackexchange.com/questions/57940/trap-int-term-exit-really-necessary) and also tested locally)
unless this script runs on dash, it should work.

Reply:
ye but on prow we have dash for example, this is exactly why some scripts / PR use SIGINT as well
btw i think for Or it didnt work on bash in one of his PRs

also you can add /bin/bash to the shebang it would be better to enforce bash for it imo

Reply:
ack, if it makes it more robust then I'm all forward
Done

Reply:
>>no, only automation/components-functests.setup.sh creates the cluster.
>
>but if that setup failed
>failed, u do want to cluster-down as some containers might already exists

automation/components-functests.setup.sh has it's own [teardown](https://github.com/kubevirt/cluster-network-addons-operator/blob/master/automation/components-functests.setup.sh#L29).

Reply:
interesting point, dont have yet an answer

wont it cluster-down when it finishes ? leaving the rest of the code without a cluster?

Reply:
it does not, otherwise the test would have failed locally

Reply:
of course, i understand, do you know why ?
afterall it has EXIT which make cluster-down, maybe in case of a success it wont run it ?
(i dont remember)

anyway lets not block on it i guess